lib/iris/aux_factory.py

```diff
@@ -767,9 +640,7 @@ def calc_bounds():
                     surface_air_pressure = surface_air_pressure_pts.reshape(
                         surface_air_pressure_pts_shape.append(1))
                 return self._derive(delta, sigma, surface_air_pressure)
-            b_shape = self._shape(nd_values_by_key)
-            b_dtype = self._dtype(nd_values_by_key)
-            bounds = _LazyArray(b_shape, calc_bounds, b_dtype)
+            bounds = calc_bounds()
```

Comment:
P.S. (following offline discussion)
Changed my mind! We could instead leave all these one-use functions in place, to emphasize the common structure of these classes in support of one day simplifying + generalising..